cmake/modules/OmrDetectSystemInformation.cmake

```diff
@@ -198,8 +198,11 @@ macro(omr_detect_system_information)
 				# just use GNU config
 				set(_OMR_TOOLCONFIG "gnu")
 			endif()
-		elseif(CMAKE_C_COMPILER_ID STREQUAL "XL" OR CMAKE_C_COMPILER_ID STREQUAL "zOS")
+		elseif(CMAKE_C_COMPILER_ID MATCHES "^XL(Clang)?$" OR CMAKE_C_COMPILER_ID STREQUAL "zOS")
 			set(_OMR_TOOLCONFIG "xlc")
+			if(CMAKE_C_COMPILER MATCHES ".*xlclang$")
+				set(CMAKE_C_COMPILER_IS_XLCLANG TRUE CACHE BOOL "XLClang is the C compiler")
```

Comment:
Is the match case-insensitive? If not, I don't see how it could match `XLClang` and `xlclang`. I suggest the patterns should be consistent.

cmake documentation isn't clear on whether matching is case-sensitive.

Reply:
If we're using xlclang, the first match serves the purpose to match the `CMAKE_C_COMPILER_ID` which will be `XL` for CMake 3.14 or prior, otherwise it will be `XLClang` for CMake 3.15+.

The second match is for `CMAKE_C_COMPILER`, which is the command to run the compiler, something like `/path/to/compiler/executable/xlclang`.

Reply:
Ah, I missed that the tests involve different variables (`CMAKE_C_COMPILER_ID` and `CMAKE_C_COMPILER`).
Maybe `CMAKE_C_COMPILER_ID` should be used both places?

Reply:
I thought maybe I could use `CMAKE_C_COMPILER_ID` in conjunction with `CMAKE_VERSION` to make it work, but if we're running with pre-CMake 3.15, it's ambiguous if we're using xlc or xlclang without checking the `CMAKE_C_COMPILER` variable. This is because pre-CMake 3.15, XLC and XLCLANG both use compiler ID `XL`. So we wouldn't be sure if we're using xlc or xlclang until we check the compiler command we're running.

Reply:
@sharon-wang could we get a summary of your previous comment right in the code? This subtlety will be non-obvious to future readers.

Reply:
Good call, I've pushed some comments to explain the above.